Library/Formula/node.rb

```diff
@@ -25,6 +25,7 @@ class Node < Formula
   option "without-completion", "npm bash completion will not be installed"
 
   depends_on :python => :build
+  depends_on :xcode => :build
```

Comment:
gyp was fixed upstream to support CLT-only. This should not be necessary. /cc @mistydemeo